Makefile

```diff
@@ -107,14 +191,89 @@ clean:
 	-rm -rf $(BASE_DIR)/coverage.out
 	@make -C examples/native clean
 
+distclean: clean
+	$(Q) rm -f .config
+	$(Q) rm -f .config.old
+
 ## check go style and static analysis
 lint:
+	$(call print_header, "Analysis source code golint & go vet")
 	$(GOLINT) ./internal/...
 	$(GOVET) -v ./internal/...
 
 ## show help
 help:
-	@make2help $(MAKEFILE_LIST)
+	@echo 'Usage: make <TARGETS>'
+	@echo ''
+	@echo 'Available targets are:'
+	@echo ''
+	@echo '    create_context     Prepare configuration.'
+	@echo '    help               Show this help screen.'
+	@echo '    clean              Remove binaries, artifacts.'
+	@echo '    test               Run unit tests.'
+	@echo '    lint               Run golint and go vet.'
+	@echo '    all                Build project for current platform.'
+	@echo ''
 
 ## define build target not a file
 .PHONY: all build test clean lint help
+
+define stop_docker_container
+	$(call print_header, "Stop Docker container")
+	-docker stop $(DOCKER_IMAGE)
+	docker ps -a
+endef
+
+##
+all:
+	make clean
+	$(call go-vendor)
+ifeq ($(CONFIG_CONTAINER),y)
+	$(call build_binary)
+	make build_docker_container
+else ifeq ($(CONFIG_NATIVE),y)
+	$(call build-object-c)
+	$(call build-result)
+else ifeq ($(CONFIG_ANDROID),y)
+	$(call print_header, "Target Binary is for Android")
+	$(call print_header, "Create Android archive from Java interface")
+	make build-object-java
+	$(call build-result)
+endif
+
+.config:
```

Comment:
Is it possible to copy the config file automatically based on the kernel information when I run "make create_context" only without `CONFIGURATION_FILE_NAME`?